lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp

```diff
@@ -2014,13 +2009,19 @@ void HdVP2Mesh::_UpdateDrawItem(
 
 #ifdef MAYA_NEW_POINT_SNAPPING_SUPPORT
     if (!isBBoxItem && !isDedicatedSelectionHighlightItem
-        && (itemDirtyBits & DirtySelectionHighlight)) {
+        && (itemDirtyBits & (DirtySelectionHighlight | DirtySelectionMode))) {
         MSelectionMask selectionMask(MSelectionMask::kSelectMeshes);
 
+#ifdef MAYA_SNAP_TO_SELECTED_OBJECTS_SUPPORT
+        if (_selectionStatus == kUnselected || drawScene.SnapToSelectedObjects()) {
```

Comment:
Updating this expression fixes the issue so that when moving a rotate pivot we can point snap to the selected objects points.